taskfiles/utils.yml

```diff
@@ -141,3 +146,194 @@ tasks:
         \( -iname "*.cpp" -o -iname "*.h" -o -iname "*.hpp" \) \
         -print0 | \
           xargs -0 clang-tidy {{.FLAGS}}
+
+  # ===
+  # CMAKE UTILS
+  # ===
+
+  # Runs cmake configure and build steps for the given source and build directories.
+  #
+  # @param {string} BUILD_DIR Cmake build directory to create.
+  # @param {string} SOURCE_DIR Project source directory containing the CMakeLists.txt file.
+  # @param {string={{.BUILD_DIR}}.md5} [CHECKSUM_FILE] Path to store the checksum of built files.
+  # @param {string=""} [CMAKE_ARGS] Any additional arguments to pass to cmake configure.
+  cmake-build:
+    label: "{{.TASK}}-{{.SOURCE_DIR}}-{{.BUILD_DIR}}"
+    internal: true
+    vars:
+      CHECKSUM_FILE: >-
+        {{default (printf "%s.md5" .BUILD_DIR) .CHECKSUM_FILE}}
+      CMAKE_ARGS: >-
+        {{default "" .CMAKE_ARGS}}
+    requires:
+      vars: ["BUILD_DIR", "SOURCE_DIR"]
+    sources:
+      - "{{.SOURCE_DIR}}/**/*"
```

Comment:
Note: Currently, this won't work for clp since the top-level CMakeLists.txt is in a directory with a lot of other files, but I guess the best solution is move all the CMakeLists and source files into a dedicated directory.

Reply:
> What do you think about changing SOURCE_DIR to SOURCE_PATHS and default CMAKELISTS_PATH to SOURCE_PATHS[0]?

Hm, would that work with `sources`? Like is there a way to pass a list variable to `sources`?

Reply:
Hmm good point... I'll look into it.